src/app/item-popup/ItemPopupBody.tsx

```diff
@@ -26,7 +26,12 @@ export default function ItemPopupBody({
   onTabChanged(tab: ItemPopupTab): void;
 }) {
   const failureStrings = Array.from(extraInfo?.failureStrings || []);
-  if (!item.canPullFromPostmaster && item.location.inPostmaster) {
+  const engramBucketHash = 375726501;
+  const isNotTransferable =
+    (item.notransfer && !item.isEngram) || (item.notransfer && !item.canPullFromPostmaster);
+  const isEngramCollected = item.location.hash === engramBucketHash;
```

Comment:
Add the following import at the top.
```suggestion
  import { BucketHashes } from 'data/d2/generated-enums';
  const isEngramCollected = item.location.hash === BucketHashes.Engrams;
```